docs/machine-learning/how-to-guides/prepare-data-ml-net.md

```diff
@@ -0,0 +1,290 @@
+    },
+    new CarData
+    {
+        Color="Blue",
+        VehicleType="Sedan"
+    },
+    new CarData
+    {
+        Color="Black",
+        VehicleType="SUV"
+    }
+};
+```
+
+The categorical `VehicleType` property can be converted into a number using the [`OneHotEncoding`](xref:Microsoft.ML.CategoricalCatalog.OneHotEncoding*) method. 
+
+```csharp
+// Define categorical transform estimator
+var categoricalEstimator = mlContext.Transforms.Categorical.OneHotEncoding("VehicleType");
+
+// Build ITransformer that performs operation
+ITransformer categoricalTransformer = categoricalEstimator.Fit(data);
+
+// Transform Data
+IDataView transformedData = categoricalTransformer.Transform(data);
+```
+
+The resulting transform converts the text value of `VehicleType` to a number. The entries in the `VehicleType` column become the following when the transform is applied: 
+
+```text
+[
+    1, // SUV
+    2, // Sedan
+    1 // SUV
+]
+```
+
+## Work with text data
+
+Text data needs to be transformed into numbers before using it to build a machine learning model. Visit the [transforms page](../resources/transforms.md) for a more detailed list and description of text transforms.
+
+Using the following input data which is loaded into an [`IDataView`](xref:Microsoft.ML.IDataView):
+
+```csharp
+ReviewData[] reviews = new ReviewData[]
+{
+    new ReviewData
+    {
+        Description="This is a good product",
+        Rating=4.7f
+    },
+    new ReviewData
+    {
+        Description="This is a bad product",
+        Rating=2.3f
+    }
+};
+```
+
+The minimum step to convert text to a numerical vector representation is to use the [`FeaturizeText`](xref:Microsoft.ML.TextCatalog.FeaturizeText*) method.
+
+```csharp
+// Define text transform estimator
+var textEstimator  = mlContext.Transforms.Text.FeaturizeText("Description");
+
+// Build ITransformer that performs operation
+ITransformer textTransformer = textEstimator.Fit(data);
+
+// Transform data
+IDataView transformedData = textTransformer.Transform(data);
+```
+
+The resulting transform would convert the text values in the `Description` column to a numerical vector that looks similar to the output below:
+
+```text
+[ 0.2041241, 0.2041241, 0.2041241, 0.4082483, 0.2041241, 0.2041241, 0.2041241, 0.2041241, 0.2041241, 0.2041241, 0.2041241, 0.2041241, 0.2041241, 0.2041241, 0.2041241, 0.2041241, 0.2041241, 0.2041241, 0.2041241, 0.2041241, 0.2041241, 0, 0, 0, 0, 0.4472136, 0.4472136, 0.4472136, 0.4472136, 0.4472136, 0 ]
+```
+
+Combine complex text processing steps to remove noise and potentially reduce the amount of required processing resources as needed.
+
+```csharp
+// Define text transform estimator
+var textEstimator = mlContext.Transforms.Text.NormalizeText("Description")
+    .Append(mlContext.Transforms.Text.TokenizeIntoWords("Description"))
+    .Append(mlContext.Transforms.Text.RemoveDefaultStopWords("Description"))
+    .Append(mlContext.Transforms.Text.FeaturizeText("Description"));
```

Comment:
I would not put those together, because the FeaturizeText does the tokenization, removing of stop words etc. 

Are you trying to give a pipeline that is equivalent to FeaturizeText? That would be a helpful thing to do, when the users are debugging their pipelines, and don;t want blackboxes.


Then you can put the following:
Text.NormalizeText
Text.TokenizeIntoWords
Text.RemoveDefaultStopWords
Text.ProduceWordBags
Text.ProduceNgrams

you'll need to concatenate the output of the last two. 

Transforms.NormalizeLpNorm

Basically do what TextFeaturizingEstimator.Fit is doing

Reply:
The one in the LDA sample is a very similar pipeline:
[https://github.com/dotnet/machinelearning/blob/master/docs/samples/Microsoft.ML.Samples/Dynamic/Transforms/Text/LatentDirichletAllocation.cs#L32](https://github.com/dotnet/machinelearning/blob/master/docs/samples/Microsoft.ML.Samples/Dynamic/Transforms/Text/LatentDirichletAllocation.cs#L32) 

  var pipeline = mlContext.Transforms.Text.NormalizeText("NormalizedText", "Text")
                .Append(mlContext.Transforms.Text.TokenizeIntoWords("Tokens", "NormalizedText"))
                .Append(mlContext.Transforms.Text.RemoveDefaultStopWords("Tokens"))
                .Append(mlContext.Transforms.Conversion.MapValueToKey("Tokens"))
                .Append(mlContext.Transforms.Text.ProduceNgrams("Tokens"))
                .Append(mlContext.Transforms.Text.LatentDirichletAllocation("Features", "Tokens", numberOfTopics: 3)); 

---
In reply to: [277379997](https://github.com/dotnet/docs/pull/11627#discussion_r277379997) [](ancestors = 277379997)